File: windmill-backend/app/models/activity.go
package models

import "time"

type Activity struct {
	Id string `json:"id"`
	Type string `json:"type"`
	Username string `json:"username"`
	UsernameF string `json:"usernamef"`
	PostId string `json:"postid"`
	Post Post `json:"post"`
	Comment Comment `json:"comment"`
	Body string `json:"body"`
	Image string `json:"image"`
	Date time.Time `json:"date"`
}

File: windmill-ios/windmill/Models/Decodable/Activity/Activity.swift
//
//  Activity.swift
//  windmill
//
//  Created by Liam  on 2020-06-23.
//  Copyright © 2020 Liam Reardon. All rights reserved.
//

import Foundation

struct Activity: Decodable {
    var id: String?
    var type: String?
    var username: String?
    var post: Post?
    var comment: Comment?
    var body: String?
    var image: String?
    
     init?(dictionary: [String: Any]) {
         
        let id = dictionary["id"] as? String
        let type = dictionary["type"] as? String
        let username = dictionary["username"] as? String
        let body = dictionary["body"] as? String
        let image = dictionary["image"] as? String
        
        self.id = id
        self.type = type
        self.username = username
        self.body = body
        self.image = image
        self.post = nil
        self.comment = nil
    
     }
}
